src/formatters/source/pushResultFormatter.ts

```diff
@@ -132,33 +133,32 @@ export class PushResultFormatter extends ResultFormatter {
     }
     // "content" property of the bundles as a string
     const contentFilePathFromDeployedBundles = this.componentsFromFilenames(
-      // the .filter(isString) should ensure only strings are present, but TS isn't finding that
-      bundlesDeployed.map((fileResponse) => fileResponse.filePath as string)
+      bundlesDeployed.map((fileResponse) => fileResponse.filePath).filter(isString)
     )
       .map((c) => c.content)
       .filter(isString);
 
     // there may be deletes not represented in the file responses (if bundle type)
     const resolver = new MetadataResolver(undefined, VirtualTreeContainer.fromFilePaths(this.deletes));
-    return (
-      this.deletes
-        .map((filePath) => {
-          const cmp = this.resolveComponentsOrWarn(filePath, resolver)[0];
```

Comment:
this [0] would be `undefined` when `resolveComponentsOrWarn` returns an empty array, which it can (and will when the filepath is empty!)